tests/xharness/Execution/MLaunchArguments.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Xharness.Execution {
+	// mlaunch is really important and has a lot of arguments that are known but
+	// used to be passed as strings. This class allows to add arguments without
+	// knowing the exact string and will also validate that an argument that
+	// needs a value does contain the valuw
+	public enum MlaunchArgumentType {
+		SdkRoot,
+		ListDev,
+		ListSim,
+		OutputFormat,
+		ListExtraData,
+	}
+
+	struct MlaunchArgument {
+		public MlaunchArgumentType Type { get; set; }
+		public string Value { get; set; }
+
+		public MlaunchArgument ((MlaunchArgumentType type, string value) data)
```

Comment:
>(MlaunchArgumentType type, string value) [](start = 26, length = 40)

This one maybe doesn't have to be a tuple

Reply:
true, no need